apps/daimo-mobile/src/view/screen/AddDeviceScreen.tsx

```diff
@@ -0,0 +1,131 @@
+import { assert } from "@daimo/common";
+import {
+  DaimoAccount,
+  DaimoNonce,
+  DaimoNonceMetadata,
+  DaimoNonceType,
+} from "@daimo/userop";
+import { BarCodeScannedCallback } from "expo-barcode-scanner";
+import { ReactNode, useMemo, useState } from "react";
+import { ActivityIndicator, View, StyleSheet } from "react-native";
+import { Hex } from "viem";
+
+import { useSendAsync } from "../../action/useSendAsync";
+import { parseAddKeyString, pubKeyToEmoji } from "../../logic/device";
+import { useAccount } from "../../model/account";
+import { getAmountText } from "../shared/Amount";
+import { ButtonBig } from "../shared/Button";
+import { Scanner } from "../shared/Scanner";
+import Spacer from "../shared/Spacer";
+import {
+  TextBold,
+  TextCenter,
+  TextError,
+  TextH2,
+  TextLight,
+} from "../shared/text";
+
+export function AddDeviceScreen() {
+  const [account] = useAccount();
+  assert(account != null);
+
+  const [newPubKey, setNewPubKey] = useState<Hex>("0x");
+  const [barCodeStatus, setBarCodeStatus] = useState<
+    "idle" | "error" | "scanned"
+  >("idle");
+
+  const handleBarCodeScanned: BarCodeScannedCallback = async ({ data }) => {
+    if (barCodeStatus !== "idle") return;
+
+    try {
+      const pubkeyHex = parseAddKeyString(data);
+      setBarCodeStatus("scanned");
```

Comment:
no need for this PR, but in future we'll likely need a way to paste in a new device key as an alternative to QR scanning